apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -77,24 +79,32 @@ async function coursesAllowedToLink({
 router.get(
   '/course_instances',
   asyncHandler(async (req, res) => {
-    const courses = await coursesAllowedToLink({
-      user_id: res.locals.authn_user.user_id,
-      is_administrator: res.locals.is_administrator,
-    });
-
     const unsafe_course_id = req.query.unsafe_course_id?.toString();
     if (!unsafe_course_id) {
       throw new HttpStatusError(400, 'Missing required parameter: unsafe_course_id');
     }
 
-    const course = courses.find((c) => c.id === unsafe_course_id);
-    if (!course) {
+    const { authzData, course } = await constructCourseOrInstanceContext({
+      user: res.locals.authn_user,
+      course_id: unsafe_course_id,
+      course_instance_id: null,
+      ip: req.ip || null,
+      req_date: res.locals.req_date,
+      is_administrator: res.locals.is_administrator,
+    });
+
+    if (!authzData || !hasRole(authzData, ['Editor'])) {
       throw new HttpStatusError(403, 'Access denied');
     }
 
     const course_instances = await courseInstancesAllowedToLink({
       course,
-      authzData: res.locals.authz_data,
+      authzData: PageAuthzDataSchema.parse({
+        ...authzData,
+        authn_user: authzData.user,
+        is_administrator: res.locals.is_administrator,
+        authn_is_administrator: res.locals.authn_is_administrator,
+      } satisfies z.input<typeof PageAuthzDataSchema>),
```

Comment:
This was necessary to make both TypeScript and runtime behavior happy. I don't like this at all, but also didn't see another way to make this work with our current suite of authz functions.

Reply:
I moved this to a helper function in e9627f2. This will be good enough for now.